client/binding.go

```diff
@@ -0,0 +1,55 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+
+	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
+	"github.com/pkg/errors"
+)
+
+// InvokeBinding invokes specific operation on the configured Dapr binding
+func (c *Client) InvokeBinding(ctx context.Context, name, op string, in []byte, inm map[string]string) (out []byte, outm map[string]string, err error) {
+	if name == "" {
+		return nil, nil, errors.New("nil topic")
+	}
+
+	req := &pb.InvokeBindingRequest{
+		Name:      name,
+		Operation: op,
+		Data:      in,
+		Metadata:  inm,
+	}
+
+	resp, err := c.protoClient.InvokeBinding(authContext(ctx), req)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "error invoking binding %s", name)
+	}
+
+	if resp != nil {
+		return resp.Data, resp.Metadata, nil
+	}
+
+	return nil, nil, nil
+}
+
+// InvokeBindingJSON invokes configured Dapr binding with an instance
+func (c *Client) InvokeBindingJSON(ctx context.Context, name, operation string, in interface{}) (out []byte, outm map[string]string, err error) {
```

Comment:
Is a separate method needed here for OutputBinding as well.?

Reply:
this method accounts for all the options (no input, no output etc)

Reply:
then along the similar lines, we dont need separate InvokeBinding & InvokeOutputBinding. One method can account for everything, thoughts?

Reply:
Yes, the one method can account for all scenarios. The only reason I left `InvokeOutputBinding(ctx context.Context, name, operation string, data []byte) error` is for simplicity if all you so is invoking a binding (most of the cases right now). As a new user, I wouldn't even know about InputBindings so the above method feels natural. Still, we can remove it if there is consensus

Reply:
Sure, lets have consistency with methods with `JSON` int he name. Either both variations should have 2 options or they both have single variation.

Reply:
It seems the JSON methods were causing confusion so I removed them in the latest commit. Resolving this comments to sync with the current code state.